src/components/InvertedFlatList/BaseInvertedFlatList.js

```diff
@@ -18,7 +18,7 @@ const propTypes = {
     initialRowHeight: PropTypes.number.isRequired,
 
     /** Passed via forwardRef so we can access the FlatList ref */
-    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({current: PropTypes.instanceOf(FlatList)})]).isRequired,
+    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({current: PropTypes.instanceOf(React.Component)})]).isRequired,
```

Comment:
Why we didn't use `PropTypes.instanceOf(BaseFlatList)` in the end?

Reply:
`It seems we are using 'React.Component' at many of the forwarded ref component. So it should be fine to replicate the same to have consistency.
I am also fine to use to this common refPropTypes created recently https://github.com/Expensify/App/blob/main/src/components/refPropTypes.js#L3`


In the approval comment I had mentioned the above https://github.com/Expensify/App/issues/20591#issuecomment-1604757259

It seems at multiple places in the app we are using the same behaviour. Kindly do let me know if we want to change this behaviour.

Reply:
hmm, I think we should try to have our prop types as tight as possible. This reference is used here for example:

https://github.com/Expensify/App/blob/70bf90d61a93a7fc8b139a4640f16436992d6728/src/libs/ReportScrollManager/index.js#L19

If you start passing around the type `Component` instead of `BaseFlatList`, typescript won't be able to find `scrollToIndex` in `Component`.

I'm not sure if this complicates things to much, then another option could be to leave this type problems for when we migrate to typescript 🤷

Reply:
Passing `Component` around when you know it is a `FlatList` (or `BaseFlatList`) feels to me like using `PropTypes.any` or `PropTypes.object` (without specifying the shape of the object)... it works... but then we are missing the purpose of types (type safety)

Reply:
If it works, I think its better to be more precise about the type, so using `FlatList` (or `BaseFlatList` if `FlatList` is not possible) is better IMO

Reply:
@aldo-expensify  then we can do the same way as done here from your example

https://github.com/Expensify/App/blob/70bf90d61a93a7fc8b139a4640f16436992d6728/src/components/InvertedFlatList/index.js#L3-L13

FlatList is imported to mention it as type. I highly doubt the `current` variable will be of type `BaseFlatList`, it will be of type `FlatList` as it was the original ref.
If we do this way when we go to typescript it won't change AFAIK.

It is not a complex change and this was also one of the proposals on the same from @allroundexperts 
It just I didn't want to add extra import for type and went with a consistent solution existing in the app.

Reply:
OOPS. Seems there where same suggestion from @allroundexperts and I also commented the same thing. (Page didn't refresh for me 😮‍💨 )

Yes. it is possible.

Reply:
Ok!, yes use `FlatList` from `react-native` if possible.